x-pack/legacy/plugins/maps/public/layers/styles/vector/vector_style.js

```diff
@@ -503,11 +503,14 @@ export class VectorStyle extends AbstractStyle {
         const dynamicStyleProp = dynamicStyleProps[j];
         const name = dynamicStyleProp.getField().getName();
         const computedName = getComputedFieldName(dynamicStyleProp.getStyleName(), name);
-        const styleValue = dynamicStyleProp.getMbValue(feature.properties[name]);
+        const rawValue = feature.properties[name];
         if (dynamicStyleProp.supportsMbFeatureState()) {
-          tmpFeatureState[computedName] = styleValue;
+          tmpFeatureState[name] = dynamicStyleProp.getNumericalMbFeatureStateValue(rawValue); //the same value will be potentially overridden multiple times, if the name remains identical
         } else {
-          feature.properties[computedName] = styleValue;
+          //in practice, a new system property will only be created for:
+          // - label text: this requires the value to be formatted first.
+          // - icon orientation: this is a lay-out property which do not support feature-state (but we're still coercing to a number)
+          feature.properties[computedName] = dynamicStyleProp.getFormattedValue(rawValue);
```

Comment:
Why is computedName still used?

Reply:
This is required for formatted values, such as labels. If Maps does not create a new property, it would override the original raw value.

Longer term, when publishing through vector tiles, we can add these system properties on the server inside the tile. Unlike the ordinal-scaling for color-ramps or styles, orientation-values or formatted labels create no cross-tile issues.